src/maps/mod.rs

```diff
@@ -890,6 +857,84 @@ fn lte_power_of_two(n: usize) -> usize {
     }
 }
 
+fn page_size() -> Result<usize, OxidebpfError> {
+    let raw_size = unsafe { libc::sysconf(libc::_SC_PAGE_SIZE) };
+
+    match raw_size.cmp(&0) {
+        std::cmp::Ordering::Less => {
+            let e = errno();
```

Comment:
So, I *think* this is fine for `_SC_PAGE_SIZE`, but generally it's possible for `sysconf` to return `-1` and not update errno for the case when you're querying a numeric value that's actually indeterminate; in that case you have to reset `errno` to `0` beforehand and see whether it has changed or not, as it might have been set at some time in the past.

I can't imagine that this would ever be a case where an indeterminate return value could happen, but I haven't checked.

Also, there's a `nix` wrapper for `sysconf` that does one layer of error handling, though it doesn't provide an amazing level of extra value here so I don't have a strong preference on whether you use it or not.

So, you don't need to change anything, but I wanted to make sure you were aware of these possibilities.